timetagger_cli/__main__.py

```diff
@@ -10,78 +12,99 @@
 # %% Some additional commands defined here
 
 
-def version():
+def version(args):
     """Print version."""
     print(f"timetagger_cli v{timetagger_cli.__version__}")
 
 
-def help():
-    """Show this help message and exit."""
-    print(docs)
-
-
 # %%
 
 
-def _make_func_dict_and_docs(*args):
-    funcs = {}
-    description = "usage: timetagger command [arguments]"
-    description += "\n\n" + timetagger_cli.__doc__.strip() + "\n\n"
-
-    for func in args:
-        if isinstance(func, str):
-            description += func + "\n\n"  # header
-        else:
-            funcs[func.__name__] = func
-            funcs[func.__name__.replace("_", "-")] = func
-            co = func.__code__
-            arg_names = " ".join(x.upper() for x in co.co_varnames[: co.co_argcount])
-            description += "    " + func.__name__ + " " + arg_names + "\n"
-            doc = "    " + func.__doc__.strip()
-            description += doc.replace("    ", "        ") + "\n"
-
-    return funcs, description
-
-
-funcs, docs = _make_func_dict_and_docs(
-    "Available commands:",
-    version,
-    help,
-    timetagger_cli.app,
-    timetagger_cli.setup,
-    timetagger_cli.status,
-    timetagger_cli.start,
-    timetagger_cli.stop,
-    timetagger_cli.resume,
-)
-
-
-def main(argv=None):
+def create_command_parser(subparsers, func):
+    parser = subparsers.add_parser(func.__name__, help=func.__doc__.strip())
+    parser.set_defaults(func=func)
+    return parser
+
+
+def setup_parser():
+    """setup argument parsing"""
+    argparser = argparse.ArgumentParser(
+        prog="timetagger", description=timetagger_cli.__doc__.strip()
+    )
```

Comment:
Could you also update the `docs` section in the readme? It just shows the result of calling `timetagger_cli` from the command line.

Reply:
done